store/postgres/src/store.rs

```diff
@@ -509,6 +521,70 @@ impl Store {
             })
     }
 
+    /// Applies an update operation to an existing entity
+    fn apply_update_operation(
```

Comment:
Apart from the added guard, I'm not sure I understand why this method is so different from `apply_set_operation`. Shouldn't this be a basic extension of `apply_set_operation`? If the actual updating and error handling logic here is better and more efficient, perhaps we should move it to `apply_set_operation`.

Reply:
The big difference is that `Update` uses the knowledge that (1) we are dealing with an existing entity and (2) which attributes to change to bring better performance. Doing this for `Set` in general would probably be worthwhile, but would require tracking more information in the mapping handlers (was the entity loaded from the db or created from scratch? which attributes were actually changed?) to get the same improvements, and would be a more invasive change.